src/ln/router.rs

```diff
@@ -112,6 +157,38 @@ impl std::fmt::Display for ChannelInfo {
 	}
 }
 
+impl Writeable for ChannelInfo {
+	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), ::std::io::Error> {
+		self.features.write(writer)?;
+		self.one_to_two.write(writer)?;
+		self.two_to_one.write(writer)?;
+		match &self.announcement_message {
```

Comment:
Same here - can we use impl_writeable?

Reply:
And there not, I mean due to DirectionalChannelInfo having itself optional field and announcement_msg being optional we have a good chunk of pattern to cover, so if macro goal is to be hygienic I'm not sure it's reached there.